pkg/distributor/distributor.go

```diff
@@ -818,12 +821,14 @@ func (d *Distributor) prePushRelabelMiddleware(next push.Func) push.Func {
 			ts := req.Timeseries[tsIdx]
 
 			if mrc := d.limits.MetricRelabelConfigs(userID); len(mrc) > 0 {
+				ts.Labels = append(ts.Labels, mimirpb.LabelAdapter{Name: metaLabelTenantID, Value: userID})
```

Comment:
This is no longer correct. `labels.Labels` must be sorted by name, and this breaks that property (on next line we convert label adapters to labels.Labels). Furthermore new `stringlabels` doesn't allow us to perform this conversion. Please revert back to using builder.

Reply:
Ah, so! Thanks for the info, reverted back.